src/main/java/bio/terra/service/snapshot/flight/create/DefaultByQueryStep.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+
+public abstract class DefaultByQueryStep implements Step {
+
+  SnapshotRequestModel getByQueryRequestModel(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
+    if (workingMap.containsKey(SnapshotWorkingMapKeys.BY_QUERY_SNAPSHOT_REQUEST_MODEL)) {
```

Comment:
There's probably a way to avoid this `if` by always setting the in-flight request model upstream. Such as, if all other validation steps copied `JobMapKeys.REQUEST` into `SnapshotWorkingMapKeys.REQUEST` or something similar.

In general I think we should try to make steps less conditional like this, as I think it's easier to reason about how a flight works if you know that a map entry is always present if a step is being run.

Reply:
I'm not sure I understand this comment fully. I agree it would be nice to make steps less conditional. The in-flight request model is always set upstream (JobMapKeys.REQUEST.getKeyName(), SnapshotRequestModel.class). That is always present. However, if the request saved there is a byRequest SnapshotRequestModel, then the convert method also saves the newly converted byQuery SnapshotRequestModel as SnapshotWorkingMapKeys.BY_QUERY_SNAPSHOT_REQUEST_MODEL. So, in that case we want to retrieve that model to use. To remove the conditional, I could replace the byRequest model saved as JobMapKeys.REQUEST.getKeyName() with the newly converted model, instead of saving it under a new key. The original ByRequestId would have already been saved under the creationInformation field on the resulting snapshot so I think it would be okay to take this approach. What do you think?